Ahayel/22 unittests

artemgael96 opened the pull request.

artemgael96 committed:
add AuthorizationControllerTests

add: AuthorizationControllerTests, AuthenticateControllerrFixture, removed this for usermanager

artemgael96 committed:
merge dev to unit

merge dev to unit

romanstat commented on exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs:
```diff
@@ -0,0 +1,50 @@
+        {
+            // Arrange
+            var _controller = new AuthenticateController(_mockUserManager.Object, _mockConfiguration.Object);
+            var loginModel = new LoginModel();
```
add email, password

romanstat commented on exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs:
```diff
@@ -0,0 +1,50 @@
+        }
+
+        [Fact]
+        public async Task FindByNameAsync_LoginModelIsEmpty_AllFieldsAreCorrect()
```
rename

romanstat commented on exalearn-api/ExaLearn.Tests/AuthenticateControllerrFixture.cs:
```diff
@@ -0,0 +1,18 @@
+{
+    public static class AuthenticateControllerrFixture
+    {
+        public static async Task<User> GetUserNameAsync() => new User() { UserName = "adminexa@mailnesia.com" };
```
password _Test1234

romanstat commented on exalearn-api/ExaLearn.Tests/AuthenticateControllerrFixture.cs:
```diff
@@ -0,0 +1,18 @@
+    {
+        public static async Task<User> GetUserNameAsync() => new User() { UserName = "adminexa@mailnesia.com" };
+
+        public static async Task<bool> IsCorrectPasswordAsync() => true;
```
(bool IsCorrect) => IsCorrect

romanstat commented on exalearn-api/ExaLearn.Tests/AuthenticateControllerTests.cs:
```diff
@@ -0,0 +1,50 @@
+            Assert.Equal(result.StatusCode, 200);
+            var token = result.Value.GetType().GetProperty("token").GetValue(result.Value);
+            Assert.NotNull(token);
+            Assert.True(Regex.IsMatch(token, @"[\S]+\.[\S]+\.[\S]"));
```
^ &

artemgael96 committed:
updated IsCorrectPasswordAsync, add FindByNameAsync_LoginModelIsInvalid_UnauthorizedResponce

updated IsCorrectPasswordAsync, add FindByNameAsync_LoginModelIsInvalid_UnauthorizedResponce
